e2e-tests/playwright/support/ui/components/channels/scheduled_draft_modal.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {expect, Locator} from '@playwright/test';
+
+export default class ScheduledDraftModal {
+    readonly container: Locator;
+
+    readonly confirmButton;
+    readonly dateInput;
+    readonly timeLocator;
+    readonly timeDropdownOptions;
+    readonly dateLocator;
+
+    constructor(container: Locator) {
+        this.container = container;
+
+        this.confirmButton = container.locator('button.confirm');
+        this.dateInput = container.locator('div.Input_wrapper');
+        this.timeLocator = container.locator('div.dateTime__input');
+        this.timeDropdownOptions = container.locator('ul.dropdown-menu .MenuItem');
+        this.dateLocator = (day: number, month: string) => container.locator(`button[aria-label*='${day}th ${month}']`);
```

Comment:
Maybe the date format here is not accurate? In some case, I'm seeing like `1st November (Friday)`

![Uploading Screenshot 2024-10-18 at 10.24.30 AM.png…]()

Reply:
Updated the locator to consider `st | nd | rd | th` for the date